test/lib/tpunit++.cpp

```diff
@@ -396,6 +403,27 @@ void tpunit::TestFixture::tpunit_detail_do_tests(TestFixture* f) {
     }
 }
 
+void tpunit::TestFixture::testLog(const string& newLog) {
+    // Format the buffer nicely as we print it out.
+    testOutputBuffer = testOutputBuffer + "[              ]    " + newLog + "\n";
+}
+
+void tpunit::TestFixture::testLog(TestFixture* f, const string& newLog) {
+    lock_guard<recursive_mutex> lock(*(f->_mutex));
+
+    // Format the buffer nicely as we print it out.
+    f->testOutputBuffer = f->testOutputBuffer + "[              ]    " + newLog + "\n";
+}
+
+void tpunit::TestFixture::flushTestBuffer(TestFixture* f) {
+    cout << f->testOutputBuffer;
+    f->testOutputBuffer = "";
+}
+
+void tpunit::TestFixture::flushTestBuffer() {
```

Comment:
I don't understand why we have two methods here. Why not just have this one with the code from the above one?